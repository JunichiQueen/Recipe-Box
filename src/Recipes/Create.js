import React, { Component } from 'react';
import axios from 'axios';

class Create extends Component{
    createRecipe = (e) => {
        e.preventDefault();
        let newRecipe = {
            name: e.target[0].value,
            description: e.target[1].value,
            ingredients: e.target[2].value,
            image: e.target[3].value
        }
        axios.post('http://localhost:5000/recipe/createRecipe', newRecipe).then(response => {
            this.props.getAll();
        }).catch(err => console.log(err));
    }

    render(){
        return(
            <div className="text-center jumbotron">
                <form onSubmit={this.createRecipe}>
                    <p>Recipename</p>
                    <input></input>
                    <p>Description</p>
                    <input></input>
                    <p>Ingredients</p>
                    <input></input>
                    <p>Image</p>
                    <input></input>
                    <button type='submit'>Create</button>
                </form>
            </div>
        )
    }

}

export default Create;