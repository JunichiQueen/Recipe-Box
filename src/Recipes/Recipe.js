import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Details from './Details.js';


function Recipe(props){

const { 
    name 
} = props;

        return(
            <div>
                    <Link to={"/"+ name}>{name}</Link>
                    <br></br>
                    

            </div>
        )
}


export default Recipe;