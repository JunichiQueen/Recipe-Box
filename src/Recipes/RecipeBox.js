import React, { Component } from 'react';
import axios from 'axios';

class RecipeBox extends Component{
    render(){
        return(
            <div class="text-center">
                <h2>
                    Recipes
                </h2>
            </div>
        )
    }

}

export default RecipeBox;