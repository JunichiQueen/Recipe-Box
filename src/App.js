import React, { Component } from 'react';
import axios from 'axios';
import './App.css';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

import RecipeBox from './Recipes/RecipeBox.js';
import Create from './Recipes/Create.js';
import Navbar from './Recipes/Navbar.js';
import Details from './Recipes/Details.js';

class App extends Component {
  constructor(){
    super();
    this.state={
      data: []
    }
  }
  getAllRecipes = () => {
    axios
    .get('http://localhost:5000/recipe/all').then(response => {
      this.setState({
        data: response.data
      })
    })
  }
  componentDidMount = () => {
    this.getAllRecipes();
  }



  render(){
    return(
      <div>
        <Navbar />
        <Router>
        
          <Route path="/" render={(props) => <RecipeBox getAll={this.getAllRecipes} data={this.state.data}/>}/>

          <Route path="/create" render={(props) => <Create getAll={this.getAllRecipes} title="A Title"/>}/>
          {this.state.data.map((item) => (
            <Route path={"/"+ item.name} render={(props) => <Details getAll={this.getAllRecipes} name={item.name} description={item.description} ingredients={item.ingredients} _id={item._id} image={item.image}/>}/>
          ))}
          
        </Router>

      </div>
    )
  }
}

export default App;
