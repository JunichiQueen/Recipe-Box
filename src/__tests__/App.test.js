import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import axios from 'axios';
import { shallow } from "enzyme";
import Navbar from '../Recipes/Navbar';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import RecipeBox from '../Recipes/RecipeBox.js';
import Create from '../Recipes/Create.js';
import renderer from 'react-test-renderer'

import { exportAllDeclaration } from '@babel/types';
import Recipe from '../Recipes/Recipe';

it('renders without crashing', () => {
  shallow(<App />);
});

it('contains navbar', () => {
  const wrapper = shallow(<App />);
  const content = <Navbar />
  expect(wrapper.contains(content)).toEqual(true);
});

it('snapshot app', () => {
  const wrapper = renderer.create(<App />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

let getAllRecipes = () => {
  axios
  .get('http://localhost:5000/recipe/all').then(response => {
    this.setState({
      data: response.data
    })
  })
}

it('render a prop', () => {
  const wrapper = shallow(<Create getAll={getAllRecipes} title="A Title"/>);
  expect(wrapper.instance().props.title).toEqual("A Title");
  expect(wrapper).toMatchSnapshot();
});
