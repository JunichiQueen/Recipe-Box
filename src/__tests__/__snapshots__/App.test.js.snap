// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a prop 1`] = `ShallowWrapper {}`;

exports[`snapshot app 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        Recipe Box
      </a>
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/Create"
              onClick={[Function]}
            >
              Create
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div
    className="text-center"
  >
    <h2>
      Recipes
    </h2>
    <table
      className="table"
    >
      <tbody />
    </table>
  </div>
</div>
`;
